engine/tests/cube-texture.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    Attribute, Color, CullMode, DrawInfo, Format, IndirectBuffer, PolygonMode,
+    PrimitiveMode, ShadeModel, ShaderStageFlagBit, Type,
+} from 'gfx/base/define';
+import { RasterizerState } from 'gfx/base/pipeline-state';
+// import { sphere } from '../../../engine/cocos/primitive';
+import { NULL_HANDLE, Program, ProgramBindings, ProgramInputs, IShaderExtension, IShaderExtensionType } from '../chassis';
+import { TestBase } from '../test-base';
+import { Vec4, Vec3, Mat4 } from '../math';
+
+export interface IGeometry {
+    positions: number[];
+    normals?: number[];
+    uvs?: number[];
+    tangents?: number[];
+    colors?: number[];
+    attributes?: Attribute[];
+    customAttributes?: {
+        attr: Attribute,
+        values: number[],
+    }[];
+
+    boundingRadius?: number;
+    minPos?: {
+        x: number;
+        y: number;
+        z: number;
+    };
+    maxPos?: {
+        x: number;
+        y: number;
+        z: number;
+    };
+    indices?: number[];
+    primitiveMode?: PrimitiveMode;
+    doubleSided?: boolean;
+}
+
+export interface ISphereOptions {
+    includeNormal: boolean;
+    includeUV: boolean;
+    segments: number;
+}
+
+export default function cube (width = 1): IGeometry {
```

Comment:
These can be imported from engine primitive module

Reply:
I tried but failed on compiling.
`import { sphere } from '../../../engine/cocos/primitive'`
![image](https://user-images.githubusercontent.com/24495793/146301499-e3d4524a-d81f-47e9-b692-9eacdd4af9a5.png)

Reply:
Try import directly from the exact source file, rather than the primitive module:
```
import box from 'primitive/box';
```